src/figuresClasses.ts

```diff
@@ -1,19 +1,107 @@
-export interface Figure {
+type Shape ='triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
 
-}
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
-export class Circle {
+  private halfSides: number;
 
+  constructor(color: string, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
```

Comment:
In TypeScript, when you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property.
So in your code, public a: number in the constructor is equivalent to writing this.a = a. You can safely remove this.